src/app/app.component.ts

```diff
@@ -7,4 +7,9 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'online-shop-angular';
+
+  @HostListener("window:beforeunload")
```

Comment:
I added it because in the description we have this: _Automatically redirect the user to the login view each time he opens or refreshes the application_